packages/web3-core/src/web3_context.ts

```diff
@@ -10,9 +10,20 @@ export class Web3Context<API extends Web3APISpec> extends Web3Config {
 
 	public readonly requestManager: Web3RequestManager<API>;
 
-	public constructor(provider: SupportedProviders<API> | string) {
+	public constructor(
+		provider: SupportedProviders<API> | string,
+		options?: Partial<ConfigOptions>,
+	) {
 		super();
 		this.requestManager = new Web3RequestManager<API>(provider);
+
+		if (options !== undefined)
+			for (const key of Object.keys(options)) {
+				// TODO
+				// @ts-expect-error Not sure how to type this
+				// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+				this[key] = options[key];
```

Comment:
This loop will not check valid keys for the config options and can cause to overrid any property or method of the context even the private one. Better to add a check for config optiosns before we override.